Algebra.java

```diff
@@ -21,47 +22,193 @@ public static void main(String args[]) {
    		System.out.println(sqrt(36));
 		System.out.println(sqrt(263169));
    		System.out.println(sqrt(76123));
-	}  
 
+		
+	}  
+	public static int abs(int x){
+		if (x<0) {
+			return -x; 	
+		}
+		return x;
+	}
+	public static int minusabs(int x){
+		if(x>0){
+			return -x;
+		}
+		return x;
+	}
 	// Returns x1 + x2
 	public static int plus(int x1, int x2) {
-		// Replace the following statement with your code
-		return 0;
+	if (x1>=0&&x2>=0) {
+		for(int i = 0 ; i<x1; i++ ){
+			x2++;
+		}
+		return x2;
 	}
+	else{
+		if (x1<0) {
+			if (x2<0) {
+				for(int i = 0 ; i>x1; i-- ){
+					x2--;
+				}
+				return x2;
+			}
+			else{
+				for(int i = 0 ; i<x2; i++ ){
+					x1++;
+				}
+				return x1;
+				}
+			}
+			else{
+				for(int i = 0 ; i<x1; i++ ){
+					x2++;
+				}
+				return x2;
+			}
+			
+		}
+	}
+	
 
 	// Returns x1 - x2
 	public static int minus(int x1, int x2) {
-		// Replace the following statement with your code
-		return 0;
-	}
+		if (x2>=0&&x1<=0) {
+			for(int i = 0 ; i<x2; i++ ){
+				x1--;
+			}
+			return x1;
+		}
+		else{
+			if (x2<0) {
+				for(int i = 0 ; i>x2; i-- ){
+					x1++;
+				}
+				return x1;
+			}
+			else  {  
+				for(int i = 0; i<x2; i++)
+				{
+					x1--;
+				}
+				return x1;
+			}
 
+			}
+		}
+	
+	
 	// Returns x1 * x2
 	public static int times(int x1, int x2) {
-		// Replace the following statement with your code
-		return 0;
+		if (x1 == 0 || x2 == 0) {
+			return 0;
+		}
+	
+		int result = 0;
+		int absX2 = Algebra.abs(x2); 
+		for (int i = 0; i < absX2; i++) {
+			result += x1; 
+		}
+		if (x2 < 0) {
+			result = -result;
+		}
+	
+		return result;
 	}
 
 	// Returns x^n (for n >= 0)
 	public static int pow(int x, int n) {
-		// Replace the following statement with your code
-		return 0;
+		int total = x;
+		int i = 1;
+		if(n != 0){
+			for (i = 1; i < n; i++) {
+				total = times(total, x);
+			}
+	    }
+		else{
+			total = 1;
+		}
+		return total;
+		
 	}
+			
+		//}
+	
 
 	// Returns the integer part of x1 / x2 
 	public static int div(int x1, int x2) {
-		// Replace the following statement with your code
-		return 0;
+		int counter = 1;
+		int result = 0;
+		if(x1==0)
+		{
+			counter = 0;
+		}
+		else{ 
```

Comment:
Again, it is not wrong, but really hard to read and understand...
It is another aspect, we as a computer scientists should consider 🤷🏻‍♂️